pkg/grpc/podstore/podstore.go

```diff
@@ -246,7 +250,12 @@ func (s store) MarkPodFailed(_ context.Context, req *podstore_protos.MarkPodFail
 	// we don't really need the CAS properties of MutateStatus but there
 	// should be only one system trying to write the status record so it
 	// doesn't hurt.
-	err = s.podStatusStore.MutateStatus(podUniqueKey, mutator)
+	trxctx, cancelFunc := transaction.New(ctx)
+	err = s.podStatusStore.MutateStatus(trxctx, podUniqueKey, mutator)
+	if err != nil {
```

Comment:
adding `defer cancelFunc()` is causing failures in both unit and integration tests...